scriptworker/cot/verify.py

```diff
@@ -887,11 +852,7 @@ def verify_task_in_task_graph(task_link, graph_defn, level=logging.CRITICAL):
     errors = []
     runtime_defn = deepcopy(task_link.task)
     # dependencies
-    # Allow for a subset of dependencies in a retriggered task.  The current use case
-    # is release promotion: we may hit the expiration deadline for a task (e.g. pushapk),
-    # and a breakpoint task in the graph may also hit its expiration.  To kick off
-    # the pushapk task, we can clone the task, update timestamps, and remove the
-    # breakpoint dependency.
```

Comment:
If I'm reading the code correctly, we still allow the actual task to have less dependencies than the one from the decision task?

Reply:
Sure. I could change this to `^` instead of `-`, or update the comment. I don't see how someone can a) keep the same taskId, b) remove some dependencies so the task has a subset of graph dependencies, and c) cause us harm. (a) and (b) seem difficult or impossible to do via the taskcluster APIs. At worst I can think of causing downstream tasks to run too early for (c).